include/vapor/ptr_cache.hpp

```diff
@@ -0,0 +1,106 @@
+//-----------------------------------------------------------------------------
+// This is an implementation of a least-recently-used (LRU) cache that keeps
+// raw pointers pointing to big structures (e.g., grids, quadtrees).
+//
+// This cache has two execution policies:
+// 1) only insertion counts as `recently used`, and
+// 2) both insertion and query count as `recently used`.
+//
+// Given this design, this cache is expected to keep the ownership of these
+// structures once they're put in the cache, and all other codes will not
+// need to manage these structures. In other words, once an object is evected,
```

Comment:
Typo: Do you mean evicted instead of evected?

Reply:
Good catch; should be fixed!